File: tadataka/vo/semi_dense/flag.py
from enum import IntEnum


class ResultFlag(IntEnum):
    SUCCESS = 0
    KEY_OUT_OF_RANGE = -1
    REF_OUT_OF_RANGE = -2
    EPIPOLAR_TOO_SHORT = -3
    INSUFFICIENT_GRADIENT = -4
    NEGATIVE_PRIOR_DEPTH = -5
    NEGATIVE_REF_DEPTH = -6
    NOT_PROCESSED = -7

File: src/semi_dense/flag.rs

#[derive(Debug)]
#[derive(PartialEq)]
pub enum Flag {
    Success = 0,
    HypothesisOutOfSerchRange = -1,
    KeyOutOfRange = -2,
    RefCloseOutOfRange = -3,
    RefFarOutOfRange = -4,
    RefEpipolarTooShort = -5,
    InsufficientGradient = -6,
    NegativePriorDepth = -7,
    NegativeRefDepth = -8,
    NotProcessed = -9
}
